File: ruby/mandelbrot.rb
BAILOUT = 16
MAX_ITERATIONS = 10000

class Mandelbrot
	
	def initialize
		output = ''
		for y in -99...99 do
			for x in -99...99 do
				i = iterate(x/100.0,y/100.0)
				if (i == 0)
					output += "*"
				else
					output += " "
				end
			end
			output += "\n"
		end
		puts output
	end

	def iterate(x,y)
		cr = y-0.5
		ci = x
		zi = 0.0
		zr = 0.0
		i = 0
		
		while(1)
			i += 1
			temp = zr * zi
			zr2 = zr * zr
			zi2 = zi * zi
			zr = zr2 - zi2 + cr
			zi = temp + temp + ci
			return i if (zi2 + zr2 > BAILOUT)
			return 0 if (i > MAX_ITERATIONS)
		end
	
	end

end

time = Time.now
Mandelbrot.new
puts
puts "Ruby #{RUBY_VERSION} Elapsed %f" % (Time.now - time)

File: c#/Mandelbrot.cs
using System;

namespace Mandelbrot {
	class Program {
		static int BAILOUT = 16;
		static int MAX_ITERATIONS = 10000;

		private static int iterate (float x, float y) {
			float cr = y - 0.5f;
			float ci = x;
			float zi = 0.0f;
			float zr = 0.0f;
			int i = 0;
			while (true) {
				i++;
				float temp = zr * zi;
				float zr2 = zr * zr;
				float zi2 = zi * zi;
				zr = zr2 - zi2 + cr;
				zi = temp + temp + ci;
				if (zi2 + zr2 > BAILOUT)
					return i;
				if (i > MAX_ITERATIONS)
					return 0;
			}
		}

		static void Main (string[] args) {
			DateTime d1 = DateTime.Now;
			int x, y;
			string output = "";

			for (y = -99; y < 99; y++) {
				for (x = -99; x < 99; x++) {
					if (iterate (x / 100.0f, y / 100.0f) == 0)
						output += "*";
					else
						output += " ";
				}
				output += "\n";
			}
			DateTime d2 = DateTime.Now;
			long diff = (long)(d2 - d1).TotalMilliseconds;
			Console.WriteLine(output + "\nC# Elapsed " + diff / 1000.0f);
		}
	}
}